// sim-node-lib/dist/Enums/ExceptionsCodesEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExceptionsCodesEnum = void 0;
var ExceptionsCodesEnum;
(function (ExceptionsCodesEnum) {
    ExceptionsCodesEnum["authorizationFaliure"] = "E_AUTHORIZATION_FAILURE";
    ExceptionsCodesEnum["badRequest"] = "E_BAD_REQUEST";
    ExceptionsCodesEnum["internalServerError"] = "E_INTERNAL_SERVER_ERROR";
    ExceptionsCodesEnum["invalidApiToken"] = "E_INVALID_API_TOKEN";
    ExceptionsCodesEnum["invalidAuthPassword"] = "E_INVALID_AUTH_PASSWORD";
    ExceptionsCodesEnum["invalidAuthUid"] = "E_INVALID_AUTH_UID";
    ExceptionsCodesEnum["notAcceptable"] = "E_NOT_ACCEPTABLE";
    ExceptionsCodesEnum["oneSessionOnly"] = "E_ONE_SESSION_ONLY";
    ExceptionsCodesEnum["routeNotFound"] = "E_ROUTE_NOT_FOUND";
    ExceptionsCodesEnum["rowNotFound"] = "E_ROW_NOT_FOUND";
    ExceptionsCodesEnum["tooManyRequests"] = "E_TOO_MANY_REQUESTS";
    ExceptionsCodesEnum["unauthorized"] = "E_UNAUTHORIZED_ACCESS";
    ExceptionsCodesEnum["validationFailed"] = "E_VALIDATION_FAILURE";
})(ExceptionsCodesEnum = exports.ExceptionsCodesEnum || (exports.ExceptionsCodesEnum = {}));
