// sim-node-lib/dist/Enums/ExceptionsMessagesEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExceptionsMessagesEnum = void 0;
var ExceptionsMessagesEnum;
(function (ExceptionsMessagesEnum) {
    ExceptionsMessagesEnum["authorizationFaliure"] = "Esta a\u00E7\u00E3o n\u00E3o foi autorizada.";
    ExceptionsMessagesEnum["internalServerError"] = "Ocorreu um erro no servidor.";
    ExceptionsMessagesEnum["invalidAuthPassword"] = "Dados enviados s\u00E3o inv\u00E1lidos!";
    ExceptionsMessagesEnum["invalidAuthUid"] = "Dados enviados s\u00E3o inv\u00E1lidos!";
    ExceptionsMessagesEnum["oneSessionOnly"] = "Essa empresa j\u00E1 possui um token ativo.";
    ExceptionsMessagesEnum["routeNotFound"] = "N\u00E3o foi encontrada nenhuma rota como essa.";
    ExceptionsMessagesEnum["rowNotFound"] = "N\u00E3o foi encontrado nenhum registro.";
    ExceptionsMessagesEnum["tooManyRequests"] = "Muitas requisi\u00E7\u00F5es foram feitas no tempo limite.";
    ExceptionsMessagesEnum["unauthorized"] = "N\u00E3o foi poss\u00EDvel autorizar ou autenticar essa requisi\u00E7\u00E3o.";
    ExceptionsMessagesEnum["validationFailed"] = "Os dados enviados s\u00E3o inv\u00E1lidos!";
})(ExceptionsMessagesEnum = exports.ExceptionsMessagesEnum || (exports.ExceptionsMessagesEnum = {}));
