// sim-node-lib/dist/Enums/PubSubTableEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PubSubTableEnum = void 0;
var PubSubTableEnum;
(function (PubSubTableEnum) {
    PubSubTableEnum["agrupador"] = "agrupador";
    PubSubTableEnum["atualiza_preco_produto_classe_cliente"] = "atualiza_preco_produto_classe_cliente";
    PubSubTableEnum["autosystem_plate"] = "autosystem-pessoa_frota";
    PubSubTableEnum["bandeira"] = "bandeira";
    PubSubTableEnum["campo"] = "campo";
    PubSubTableEnum["cargo"] = "cargo";
    PubSubTableEnum["cidade"] = "cidade";
    PubSubTableEnum["cidadeAgrupadorCidade"] = "cidade_agrupador_cidade";
    PubSubTableEnum["cidadeAgrupador"] = "cidade_agrupador";
    PubSubTableEnum["classe"] = "classe";
    PubSubTableEnum["cliente"] = "cliente";
    PubSubTableEnum["clienteEndereco"] = "cliente_endereco";
    PubSubTableEnum["clientePendente"] = "cliente_pendente";
    PubSubTableEnum["configuracao"] = "configuracao";
    PubSubTableEnum["concorrente"] = "concorrente";
    PubSubTableEnum["empresa"] = "empresa";
    PubSubTableEnum["empresaAgrupador"] = "empresa_agrupador";
    PubSubTableEnum["empresaConcorrente"] = "empresa_concorrente";
    PubSubTableEnum["grupoEmpresaEmpresa"] = "grupo_empresa_empresa";
    PubSubTableEnum["empresaCargoUsuario"] = "empresa_cargo_usuario";
    PubSubTableEnum["empresaHorario"] = "empresa_horario";
    PubSubTableEnum["empresaItem"] = "empresa_item";
    PubSubTableEnum["empresaServico"] = "empresa_servico";
    PubSubTableEnum["formaPagamento"] = "forma_pagamento";
    PubSubTableEnum["frotaProduto"] = "frota_produto";
    PubSubTableEnum["grupoDesconto"] = "grupo_desconto";
    PubSubTableEnum["grupoDescontoLimite"] = "grupo_desconto_limite";
    PubSubTableEnum["grupoPromocao"] = "grupo_promocao";
    PubSubTableEnum["grupoEmpresa"] = "grupo_empresa";
    PubSubTableEnum["item"] = "item";
    PubSubTableEnum["itemTag"] = "item_tag";
    PubSubTableEnum["modeloPredicao"] = "modelo_predicao";
    PubSubTableEnum["pessoaFrotaLimite"] = "pessoa_frota_limite";
    PubSubTableEnum["pessoaFrotaProduto"] = "pessoa_frota_produto";
    PubSubTableEnum["placas"] = "placas";
    PubSubTableEnum["premiacao"] = "premiacao";
    PubSubTableEnum["premiacaoCombo"] = "premiacao_combo";
    PubSubTableEnum["premiacaoComboItem"] = "premiacao_combo_item";
    PubSubTableEnum["premiacaoGrupoDesconto"] = "premiacao_grupo_desconto";
    PubSubTableEnum["premiacaoPeriodo"] = "premiacao_periodo";
    PubSubTableEnum["premiacaoPeriodoRecorrencia"] = "premiacao_periodo_recorrencia";
    PubSubTableEnum["premiacaoCidade"] = "premiacao_cidade";
    PubSubTableEnum["promocao"] = "promocao";
    PubSubTableEnum["promocaoCidade"] = "promocao_cidade";
    PubSubTableEnum["promocaoCombo"] = "promocao_combo";
    PubSubTableEnum["promocaoComboItem"] = "promocao_combo_item";
    PubSubTableEnum["promocaoComboTag"] = "promocao_combo_tag";
    PubSubTableEnum["promocaoEmpresa"] = "promocao_empresa";
    PubSubTableEnum["promocaoGrupoDesconto"] = "promocao_grupo_desconto";
    PubSubTableEnum["promocaoRecorrencia"] = "promocao_recorrencia";
    PubSubTableEnum["proposta_classe_retorno_precificacao"] = "proposta_classe_retorno_precificacao";
    PubSubTableEnum["proposta_preco_envio_autosystem"] = "proposta_de_preco_envio_autosystem";
    PubSubTableEnum["proposta_preco_retorno_autosystem"] = "proposta_de_preco_retorno_autosystem";
    PubSubTableEnum["rede"] = "rede";
    PubSubTableEnum["regra_preco_cliente_autosystem"] = "regra_preco_cliente_autosystem";
    PubSubTableEnum["remove_preco_produto_classe_cliente"] = "remove_preco_produto_classe_cliente";
    PubSubTableEnum["status_processo_update_preco_produto"] = "status_processo_update_preco_produto";
    PubSubTableEnum["servico"] = "servico";
    PubSubTableEnum["sorteio"] = "sorteio";
    PubSubTableEnum["sorteioCliente"] = "sorteio_cliente";
    PubSubTableEnum["sorteioResultado"] = "sorteio_resultado";
    PubSubTableEnum["tag"] = "tag";
    PubSubTableEnum["tipoPremiacao"] = "tipo_premiacao";
    PubSubTableEnum["voucherDesconto"] = "voucher_desconto";
    PubSubTableEnum["voucherDescontoLote"] = "voucher_desconto_lote";
    PubSubTableEnum["voucherIndividual"] = "voucher_individual";
    PubSubTableEnum["voucherMigracao"] = "voucher_migracao";
    PubSubTableEnum["voucherMigracaoLote"] = "voucher_migracao_lote";
})(PubSubTableEnum = exports.PubSubTableEnum || (exports.PubSubTableEnum = {}));
