// sim-node-lib/dist/Models/PriceToSale.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Origem = exports.Item = exports.Preco = exports.Empresa = exports.Grupo = exports.PriceToSale = void 0;
class PriceToSale {
    constructor(grupo, empresa, precos) {
        this.grupo = grupo;
        this.empresa = empresa;
        this.precos = precos;
    }
}
exports.PriceToSale = PriceToSale;
class Grupo {
    constructor(codigo, descricao) {
        this.codigo = codigo;
        this.descricao = descricao;
    }
}
exports.Grupo = Grupo;
class Empresa {
    constructor(cnpj, key, descricao, cidade, estado, latitude, longitude) {
        this.cnpj = cnpj;
        this.key = key;
        this.descricao = descricao;
        this.cidade = cidade;
        this.estado = estado;
        this.latitude = latitude;
        this.longitude = longitude;
    }
}
exports.Empresa = Empresa;
class Preco {
    constructor(item, preco_base, preco_final, desconto, origem, ordem, aplicavel, herdou_default, arredondamento_psicologico) {
        this.item = item;
        this.preco_base = preco_base;
        this.preco_final = preco_final;
        this.desconto = desconto;
        this.origem = origem;
        this.ordem = ordem;
        this.aplicavel = aplicavel;
        this.herdou_default = herdou_default;
        this.arredondamento_psicologico = arredondamento_psicologico;
    }
}
exports.Preco = Preco;
class Item {
    constructor(codigo_erp, descricao, key) {
        this.codigo_erp = codigo_erp;
        this.descricao = descricao;
        this.key = key;
    }
}
exports.Item = Item;
class Origem {
    constructor(politica, regra_key) {
        this.politica = politica;
        this.regra_key = regra_key;
    }
}
exports.Origem = Origem;
